libafl_targets/src/sancov_pcguard.rs

```diff
@@ -22,11 +22,11 @@ pub unsafe extern "C" fn __sanitizer_cov_trace_pc_guard(guard: *mut u32) {
     {
         #[cfg(feature = "sancov_pcguard_edges")]
         {
-            (EDGES_MAP_PTR as *mut u8).add(pos).write(1);
```

Comment:
I'm not sure if this is right. I'll check tomorrow

Reply:
(I don't trust clippy when they're suggesting something in unsafe code)

Reply:
I'll test it and find out

Reply:
for this part I think is ok.
but I just am doubtful about clippy's suggestion on the code touching pointers.

because once they really suggested something that does wrong & doesn't even compile
https://rust-lang.github.io/rust-clippy/master/index.html#/ptr_cast_constness?versions=gte:71

Reply:
Ok I checked, with:

main.rs:
```rust
extern "C" {
    static mut __afl_area_ptr: *mut u8;
}
use __afl_area_ptr as EDGES_MAP_PTR;

unsafe fn __sanitizer_cov_trace_pc_guard(guard: *mut u32) {
    let pos = *guard as usize;

    let x = (unsafe { EDGES_MAP_PTR }).add(pos);
    let y = (unsafe { EDGES_MAP_PTR as *mut u8 }).add(pos);
    println!("x={:#x} y={:#x}", x as usize, y as usize);
    x.write(1);
    println!("*x={:#x}", *x);
    y.write(2);
    println!("*y={:#x}", *x);
}

fn main() {
    let mut guard_pos = 8u32;
    let guard = &mut guard_pos as *mut u32;
    unsafe { __sanitizer_cov_trace_pc_guard(guard) };
}
```

test.c (`clang -c -o test.o test.c && llvm-ar rc target/debug/libtest.a test.o`)
```
#include <stddef.h>
#include <stdint.h>

#define EDGES_MAP_SIZE (65536)

uint8_t __afl_area[EDGES_MAP_SIZE] = {0};
uint8_t *__afl_area_ptr = &__afl_area[0];
```

output:
```
x=0x560be87cd048 y=0x560be87cd048
*x=0x1
*y=0x2
```